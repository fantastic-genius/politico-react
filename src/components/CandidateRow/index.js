import React from 'react';
import userImg from '../../assets/images/user.jpg'

const CandidateRow = ({candidate, onVote}) => {
  const {
    id,
    firstname,
    lastname,
    passportUrl,
    partyname,
    logourl
  } = candidate;

  /* istanbul ignore handleCheck */
  const handleCheck = (e) => {
    /* istanbul ignore next */
    const choice = e.target
    /* istanbul ignore next */
    choice.checked = true;
    /* istanbul ignore next */
    const choices = document.querySelectorAll('.choice');
    /* istanbul ignore next */
    let parentTr = choice.closest('tr');
    /* istanbul ignore next */
    let corBtn = parentTr.querySelectorAll('button');
    /* istanbul ignore next */
    corBtn[0].classList.remove('hide');

    /* istanbul ignore next */
    choices.forEach(choic => {
      /* istanbul ignore if */
      if(choic !== choice){
        choic.checked = false;
        parentTr = choic.closest('tr');
        corBtn = parentTr.querySelectorAll('.btn');
        if(!corBtn[0].classList.contains('hide')){
            corBtn[0].classList.add('hide');
        }
      }
    })
  }

  return (
    <tr>
      <td><input type="checkbox" className="choice" name="choice" onClick={handleCheck} /></td>
      <td><img src={passportUrl || userImg} alt={firstname} /></td>
      <td>{`${firstname} ${lastname}`}</td>
      <td>{partyname}</td>
      <td><img src={logourl} alt="Logo" /></td>
      <td>
        <button className="btn btn-warning hide" data-candidateid={id} onClick={onVote}>Vote</button>
      </td>
    </tr>
  );
};

export default CandidateRow;
