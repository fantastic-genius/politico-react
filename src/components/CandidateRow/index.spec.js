import React from 'react';
import { shallow } from 'enzyme';
import CandidateRow from './index';

describe('<CandidateRow />', () => {
  describe('Rendering', () => {
    it('should render CandidateRow correctly', () => {
      const candidate = {
        id: 1,
        firstname: 'john',
        lastname: 'mark',
        passportUrl: 'https://w.pe./img',
        partyname: 'All peoples party',
        logourl: 'https://app.com/logo'
      }
      const mockOnVote = jest.fn(() => {});
      const component = shallow(<CandidateRow candidate={candidate} onVote={mockOnVote} />);
      expect(component).toMatchSnapshot();
    });
  })

})

