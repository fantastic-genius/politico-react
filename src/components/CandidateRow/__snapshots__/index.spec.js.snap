// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CandidateRow /> Rendering should render CandidateRow correctly 1`] = `
<tr>
  <td>
    <input
      className="choice"
      name="choice"
      onClick={[Function]}
      type="checkbox"
    />
  </td>
  <td>
    <img
      alt="john"
      src="https://w.pe./img"
    />
  </td>
  <td>
    john mark
  </td>
  <td>
    All peoples party
  </td>
  <td>
    <img
      alt="Logo"
      src="https://app.com/logo"
    />
  </td>
  <td>
    <button
      className="btn btn-warning hide"
      data-candidateid={1}
      onClick={[MockFunction]}
    >
      Vote
    </button>
  </td>
</tr>
`;
